scripts/update-index-alias.py

```diff
@@ -0,0 +1,111 @@
+#!/usr/bin/env python
+"""
+A script to update the search index an alias is assoiciated with.
+
+To delete the old index (the index losing the '<alias>-old' alias), set --delete-old-index=yes
+
+Usage:
+scripts/update-index-alias.py <alias> <target> <stage> <search-api-endpoint> [options]
+
+Options:
+    --delete-old-index=<delete-old-index>  [default: no]
+"""
+import os
+import sys
+import subprocess
+import yaml
+import json
+from distutils.util import strtobool
+import requests
+from requests.exceptions import HTTPError
+from docopt import docopt
+
+
+def update_index_alias(alias, target, stage, endpoint, delete_old_index):
+    auth_token = _get_auth_token(stage)
+    headers = {
+        'Authorization': "Bearer {}".format(auth_token),
+        'Content-type': 'application/json'
+    }
+
+    if delete_old_index:
+        old_index_to_delete = _get_old_index_to_delete(alias, endpoint)
+
+    _rename_current_alias_to_old(alias, endpoint, headers)
+
+    if delete_old_index:
+        _delete_old_index(old_index_to_delete, endpoint, auth_token)
```

Comment:
So if this step fails on first run and interrupts the script, then we're left with `alias-old == alias`. If the script is run again it will try to move `alias-old` to match `alias` (which it already does) and then retry the delete, which would fail (since search-api doesn't allow deleting indexes with an alias). The only way to resolve this would be manual intervention to move the aliases and clean up the old index.

I suggest we reorder the steps:
1. Store the currently aliased index name
2. Store the old aliased index name
3. Move the alias to the new index
4. Move alias-old to the previously aliased index
5. Delete previous target of alias-old

This way we should be able to rerun the job after failure and the worst case is that it will leave some old indexes around.